pkg/client/cache/listwatch.go

```diff
@@ -60,9 +61,7 @@ func NewListWatchFromClient(c Getter, resource string, namespace string, fieldSe
 			Prefix("watch").
 			Namespace(namespace).
 			Resource(resource).
-			// TODO: Use VersionedParams once this is supported for non v1 API.
-			Param("resourceVersion", options.ResourceVersion).
-			TimeoutSeconds(timeoutFromListOptions(options)).
+			VersionedParams(&options, api.Scheme).
```

Comment:
Ah, I think I know why we named it VersionedParams, because we really have "versioned" options like PodLogOptions. I forgot the context of this PR after the holidays. Sorry for the noise, please ignore my previous comment.